libraries/AP_HAL_Linux/CAN.h

```diff
@@ -0,0 +1,213 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "AP_HAL_Linux.h"
+#include <AP_HAL/CAN.h>
+
+#include <linux/can.h>
+
+#include <string>
+#include <queue>
+#include <memory>
+#include <map>
+#include <unordered_set>
+#include <poll.h>
+
+namespace Linux {
+
+enum class SocketCanError
+{
+    SocketReadFailure,
+    SocketWriteFailure,
+    TxTimeout
+};
+
+class LinuxCAN: public AP_HAL::CAN {
```

Comment:
This is already inside the Linux namespace. We moved away from "Linux::Linux*" naming some time ago. I think this should be named simply CAN.